ext/B/B.xs

```diff
@@ -1422,6 +1422,30 @@ object_2svref(sv)
     PPCODE:
 	ST(0) = sv_2mortal(newRV(sv));
 	XSRETURN(1);
+
+bool
+is_bool(sv)
+    B::SV   sv
+    CODE:
+    RETVAL = SvIsBOOL(sv) ? TRUE : FALSE;
```

Comment:
Why the ternary? This can simply be

```
RETVAL = SvIsBOOL(sv);
```

Ditto the other two functions below.

Reply:
Actually for that matter, why does it have its own name? Why can it not simply be the SvIsBOOL itself?

```
bool
SvIsBOOL(sv)
    B::SV  sv
```

with no code body at all. E.g. compare to the `SvIV` below.

Reply:
that could work

Reply:
note that this should not use `SvIsBool` as there is no prefix for the `B::SV` package set
so I gave a try of using 
```
bool
IsBOOL(sv)
    B::SV   sv
```

this is now failing with
```
./perl: symbol lookup error: ../lib/auto/B/B.so: undefined symbol: IsBOOL
```

note sure how to fix it. If you have any ideas or suggestion I'm glad to take it
thanks

Reply:
ok need to setup the prefix... but then this become a mess
```diff
diff --git a/ext/B/B.xs b/ext/B/B.xs
index 878fb0dba5..4f75bec4f0 100644
--- a/ext/B/B.xs
+++ b/ext/B/B.xs
@@ -1398,12 +1398,12 @@ aux_list(o, cv)



-MODULE = B     PACKAGE = B::SV
+MODULE = B     PACKAGE = B::SV         PREFIX = Sv

 #define MAGICAL_FLAG_BITS (SVs_GMG|SVs_SMG|SVs_RMG)

 U32
-REFCNT(sv)
+SvREFCNT(sv)
        B::SV   sv
     ALIAS:
        FLAGS = 0xFFFFFFFF
@@ -1417,7 +1417,7 @@ REFCNT(sv)
        RETVAL

 void
-object_2svref(sv)
+Svobject_2svref(sv)
        B::SV   sv
     PPCODE:
        ST(0) = sv_2mortal(newRV(sv));
@@ -1428,7 +1428,7 @@ SvIsBOOL(sv)
     B::SV   sv

 bool
-is_bool_yes(sv)
+Svis_bool_yes(sv)
     B::SV   sv
     CODE:
     RETVAL = SvIsBOOL(sv) && SvTRUE(sv);
@@ -1436,7 +1436,7 @@ is_bool_yes(sv)
         RETVAL

 bool
-is_bool_no(sv)
+Svis_bool_no(sv)
     B::SV   sv
     CODE:
     RETVAL = SvIsBOOL(sv) && !SvTRUE(sv);
```

Reply:
Would you really prefer this flavor?

Reply:
Using PREFIX generally only makes sense if you don't have a CODE body